pkg/clientcert/cert_controller.go

```diff
@@ -166,7 +205,47 @@ func (c *clientCertificateController) sync(ctx context.Context, syncCtx factory.
 
 	// reconcile pending csr if exists
 	if len(c.csrName) > 0 {
-		newSecretConfig, err := c.syncCSR()
+		// build a secret data map if the csr is approved
+		newSecretConfig, err := func() (map[string][]byte, error) {
+			// skip if there is no ongoing csr
+			if len(c.csrName) == 0 {
+				return nil, fmt.Errorf("no ongoing csr")
+			}
+
+			csr, err := c.csrControl.get(c.csrName)
+			if err != nil {
```

Comment:
if the csr is not found, why we remove the old fallback logic here? has some special reasons or we should keep the logic